api/proto/config/storage.proto

```diff
@@ -242,3 +243,48 @@ message DisksConfig {
   DisksArrayType array_type = 2;
   repeated DisksConfig children = 3;
 }
+
+// SnapshotOpsCmd counter for snapshot cmd
+message SnapshotOpsCmd {
+  uint32 counter = 1;
+  google.protobuf.Timestamp opsTime = 2;
+}
+
+// ZfsSnapshot describes a snapshot for a specific logical
+// volume that must exist on the device. It has a required
+// field volume_uuid for which it was created.
+message ZfsSnapshot {
+  // The real snapshot name in ZFS
+  string id = 1;
+
+  // Display name (User-friendly name). This name does not affect
+  // the snapshot properties in ZFS in any way
+  string display_name = 2;
+
+  // ID of the volume this snapshot belongs to. The field is required
+  // for all commands except for the rename command.
+  string volume_uuid = 3;
+
+  // Time of the last operation on the snapshot
+  uint64 last_cmd_timestamp = 4;
```

Comment:
Can you please describe the logic of having timestamp here?
My understanding here: if we want to create snapshot, we just add ZfsSnapshot object, if we want to delete, we remove ZfsSnapshot object from list. We can use id as an internal "name" of snapshot inside zfs and to not create them twice. So, all what we need is rollback cmd (counter increment) that will indicate that we want to rollback volume (inside volume_uuid) onto this snapshot (we can internally found it by id). Please correct me if I am wrong or better describe your logic somewhere.